app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,382 @@
+    progress_bar_searching_peers.visibility = View.GONE
+    list_peer_devices.visibility = View.VISIBLE
+    wifiPeerListAdapter?.items = deviceList
+    if (deviceList.isEmpty()) {
+      Log.d(TAG, "No devices found")
+    }
+  }
+
+  override fun onFileTransferComplete() {
+    finish()
+  }
+
+  /* Helper methods used for checking permissions and states of services */
+  private fun checkCoarseLocationAccessPermission(): Boolean {
+    // Required by Android to detect wifi-p2p peers
+    return if (ContextCompat.checkSelfPermission(
+        this,
+        Manifest.permission.ACCESS_COARSE_LOCATION
+      )
+      == PackageManager.PERMISSION_DENIED
+    ) {
+      if (ActivityCompat.shouldShowRequestPermissionRationale(
+          this,
+          Manifest.permission.ACCESS_COARSE_LOCATION
+        )
+      ) {
+        alertDialogShower.show(KiwixDialog.LocationPermissionRationale,
+          {
+            ActivityCompat.requestPermissions(
+              this, arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION),
+              PERMISSION_REQUEST_CODE_COARSE_LOCATION
+            )
+          })
+      } else {
+        ActivityCompat.requestPermissions(
+          this, arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION),
+          PERMISSION_REQUEST_CODE_COARSE_LOCATION
+        )
+      }
+      false
+    } else {
+      true
+      // Control reaches here: Either permission granted at install time, or at the time of request
+    }
+  }
+
+  private fun checkExternalStorageWritePermission(): Boolean { // To access and store the zims
+    return if (ContextCompat.checkSelfPermission(
+        this,
+        Manifest.permission.WRITE_EXTERNAL_STORAGE
+      )
+      == PackageManager.PERMISSION_DENIED
+    ) {
+      if (ActivityCompat.shouldShowRequestPermissionRationale(
+          this,
+          Manifest.permission.WRITE_EXTERNAL_STORAGE
+        )
+      ) {
+        alertDialogShower.show(KiwixDialog.StoragePermissionRationale, {
+          ActivityCompat.requestPermissions(
+            this@LocalFileTransferActivity,
+            arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+            PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS
+          )
+        })
+      } else {
+        ActivityCompat.requestPermissions(
+          this, arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+          PERMISSION_REQUEST_CODE_STORAGE_WRITE_ACCESS
+        )
+      }
+      false
+    } else {
+      true
+      // Control reaches here: Either permission granted at install time, or at the time of request
+    }
+  }
+
+  override fun onRequestPermissionsResult(
+    requestCode: Int,
+    permissions: Array<String>,
+    grantResults: IntArray
+  ) {
+    if (grantResults[0] == PackageManager.PERMISSION_DENIED) {
+      when (requestCode) {
+        PERMISSION_REQUEST_CODE_COARSE_LOCATION -> {
+          Log.e(
+            TAG,
+            "Location permission not granted"
+          )
```

Comment:
Same goes for concatenation here. Note that: if some loc(lines of code) can be fit in a single line then please do that, but do note that, NO Semicolons are allowed on a single line